src/components/header/modals/ModalPolkasafe.vue

```diff
@@ -0,0 +1,402 @@
+      await props.disconnectAccount();
+      isClosing.value = true;
+      const animationDuration = 500;
+      await wait(animationDuration);
+      store.commit('general/setCurrentAddress', substrateAccount);
+      localStorage.setItem(LOCAL_STORAGE.SELECTED_WALLET, SupportMultisig.Polkasafe);
+      const multisigObj = {
+        multisigAccount: multisigAccounts.value.find((it) => it.address === substrateAccount),
+        signatory: selectedSignatory.value,
+      } as Multisig;
+      localStorage.setItem(LOCAL_STORAGE.MULTISIG, JSON.stringify(multisigObj));
+      isSelected.value = true;
+      isClosing.value = false;
+      emit('update:is-open', false);
+      window.dispatchEvent(new CustomEvent(LOCAL_STORAGE.SELECTED_WALLET));
+    };
+
+    const selAccount = ref<string>('');
+    const subScan = computed<string>(
+      () => `${providerEndpoints[currentNetworkIdx.value].subscan}/account/`
+    );
+
+    const previousSelIdx = computed<number | null>(() => {
+      if (substrateAccounts.value && props.currentAccount) {
+        const index = substrateAccounts.value.findIndex(
+          (it: SubstrateAccount) => it.address === props.currentAccount
+        );
+        return index;
+      } else {
+        return null;
+      }
+    });
+
+    const copyAddress = (address: string): void => {
+      copy(address);
+      store.dispatch('general/showAlertMsg', {
+        msg: t('toast.copyAddressSuccessfully'),
+        alertType: 'copied',
+      });
+    };
+
+    const windowHeight = ref<number>(window.innerHeight);
+    const onHeightChange = (): void => {
+      const adjustment = width.value > screenSize.sm ? 520 : 390;
+      windowHeight.value = window.innerHeight - adjustment;
+    };
+
+    window.addEventListener('resize', onHeightChange);
+    onHeightChange();
+
+    const displayBalance = (balance: string): number => {
+      return truncate(ethers.utils.formatEther(balance || '0'));
+    };
+
+    const handleGetMultisigAccounts = async (): Promise<void> => {
+      multisigAccounts.value = [];
+      isLoadingPolkasafe.value = true;
+      selAccount.value = '';
+      if (!selectedSignatory.value || currentNetworkChain.value !== astarChain.ASTAR) {
+        return;
+      }
+      try {
+        const extensions = await web3Enable('AstarNetwork/astar-apps');
+        const injector = extensions.find((it) => {
+          return selectedSignatory.value && it.name === selectedSignatory.value.source;
+        }) as any;
+
+        if (!injector) {
+          throw Error('injector not found');
+        }
+        const substratePrefix = 42;
+        const signatory = encodeAddress(selectedSignatory.value.address, substratePrefix);
+
+        const setMultisigAccounts = async (c: Polkasafe): Promise<void> => {
+          try {
+            const { data, error } = await c.connectAddress(signatory);
+            if (error) throw Error(error);
```

Comment:
Throwing error doesn't help here because it will be handled by catch below and written to console

Reply:
@bobo-k2  I think that logging the error on the console is enough for now. So there's no need to run the rest of the 'try' code in case an error occurs (though I haven't encountered the error).

Reply:
Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/834/commits/e616ee355c3369c39ee55bf2eaa960923447d482) commit.